linkerd/io/src/boxed.rs

```diff
@@ -10,14 +10,14 @@ pub struct BoxedIo(Pin<Box<dyn Io + Unpin>>);
 /// This is necessary for `BoxedIo`, as `dyn AsyncRead + AsyncWrite + PeerAddr`
 /// is not a valid trait object. However, it needn't be public --- it's just
 /// used internally.
-trait Io: AsyncRead + AsyncWrite + PeerAddr + Send {}
+trait Io: AsyncRead + AsyncWrite + PeerAddr + Send + Sync {}
```

Comment:
what necessitated the `Sync` bound?

Reply:
We need this so that boring's error implements `Error`
```
error[E0277]: `I` cannot be shared between threads safely
   --> linkerd/meshtls/boring/src/server.rs:68:66
    |
68  |                     None => io::Error::new(io::ErrorKind::Other, e),
    |                                                                  ^ `I` cannot be shared between threads safely
    |
    = note: required because it appears within the type `tokio_boring::StreamWrapper<I>`
    = note: required because it appears within the type `PhantomData<tokio_boring::StreamWrapper<I>>`
    = note: required because it appears within the type `boring::ssl::SslStream<tokio_boring::StreamWrapper<I>>`
    = note: required because it appears within the type `MidHandshakeSslStream<tokio_boring::StreamWrapper<I>>`
    = note: required because it appears within the type `boring::ssl::HandshakeError<tokio_boring::StreamWrapper<I>>`
    = note: required because it appears within the type `tokio_boring::HandshakeError<I>`
    = note: required because of the requirements on the impl of `From<tokio_boring::HandshakeError<I>>` for `Box<(dyn std::error::Error + Send + Sync + 'static)>`
    = note: required because of the requirements on the impl of `Into<Box<(dyn std::error::Error + Send + Sync + 'static)>>` for `tokio_boring::HandshakeError<I>`
note: required by a bound in `linkerd_io::Error::new`
help: consider further restricting type parameter `I`
    |
49  |     I: std::fmt::Debug, I: std::marker::Sync // Needed so that boring's error type implements Error.
    |                       ~~~~~~~~~~~~~~~~~~~~~~
```